### DISCUSSION TITLE
More capable permission filter rules

### DISCUSSION THREAD

**Discussion by InterFelix** [directus/directus#10927] [Feature Requests]
I am trying to build an app that has different "spaces" for different groups of people.
So when one person logs in, they see the space they're assigned to, and when someone else logs in, they see a different space, the on they're assigned to.
No I'm trying to work out how to best do this.
My first idea was to have a collection of "spaces" and store the ID of their assigned space in the user collection. 
Then have all the other collections relate to the "spaces" collection through a foreign key and create a role that allows any user access to only those items in any given collection that belong to the space they are assigned to.
My problem is the following:
When I go to create a permission, I can only create rules that compare field values against a value hardcoded in this rule.
What I would like to do is compare a field value of one item with a field value in another item (the system user item in this case).

tl;dr:
I would like filter rules of custom permissions to be able to compare field values from two different collections, not only a field value in one collection with a hardcoded value.

**Comment by AlphaMoonbaseBerlin** [comment]
You can use $CURRENT_USER to tget a refference to the logged in user. So when you add a relation to "spaces" to user you can check against $CURRENT_USER.spaces and check if the space-field of the current item is in $CURRENT_USER.spaces

**Comment by InterFelix** [comment]
Thank you so much for this valuable suggestion! This helps me a ton, actually.
However, I have discovered one limitation - it doesn't seem to support Many-to-Many-Relationships.
The only comparison operator that sounds vaguely suitable is "is one of", but that one expects a list of values to compare with.
If I put `$CURRENT_USER.spaces` into one of the value fields, it just doesn't work.

So I have a new suggestion: Add Support for M2M-Relationship fields in Permission rule filter operators

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
